designcompose/src/main/java/com/android/designcompose/squoosh/SquooshRoot.kt

```diff
@@ -478,6 +417,110 @@ fun SquooshRoot(
                     animationValues,
                 ),
             measurePolicy = { measurables, constraints ->
+                // Update the root node style to have the incoming width/height from our parent
+                // Composable.
+                root.applyLayoutConstraints(constraints, density)
+
+                // Perform layout on our resolved tree.
+                layoutTree(root, rootLayoutId, layoutIdAllocator, layoutCache, layoutValueCache)
```

Comment:
This is the main point of the change; evaluating layout happens in Compose's `measure` phase. That means we have the incoming constraints already, which are applied to the root's style on L422, so we can do layout in one phase.